clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/ui/sourceFoldersView/UTBotProjectViewPaneForSettings.kt

```diff
@@ -3,12 +3,13 @@ package org.utbot.cpp.clion.plugin.ui.sourceFoldersView
 import com.intellij.ide.projectView.impl.AbstractProjectTreeStructure
 import com.intellij.ide.projectView.impl.ProjectViewTree
 import com.intellij.openapi.project.Project
-import javax.swing.tree.DefaultTreeModel
+import org.utbot.cpp.clion.plugin.settings.settings
 import org.utbot.cpp.clion.plugin.ui.wizard.steps.ObservableValue
-import org.utbot.cpp.clion.plugin.utils.utbotSettings
+import javax.swing.tree.DefaultTreeModel
 
 open class UTBotProjectViewPaneForSettings(project: Project) : UTBotProjectViewPane(project) {
     private val sourceDirs: ObservableValue<Set<String>> = initObservableDirectories()
+    private val settings = myProject.settings.storedSettings
```

Comment:
Here you store reference to state of the `UTBotProjectStoredSettings.State`, which can be changed by the ide (e.g. when ide loads state from xml file) making this reference outdated. So I suggest to use: 
`private val settings: UTBotProjectStoredSettings.State get() = myProject.settings.storedSettings`